sdk/logs/src/main/java/io/opentelemetry/sdk/logs/LoggerSharedState.java

```diff
@@ -18,28 +18,29 @@
  */
 final class LoggerSharedState {
   private final Object lock = new Object();
-  private final Resource resource;
+  private final Supplier<Resource> resourceSupplier;
   private final Supplier<LogLimits> logLimitsSupplier;
   private final LogRecordProcessor logRecordProcessor;
   private final Clock clock;
   private final ExceptionAttributeResolver exceptionAttributeResolver;
   @Nullable private volatile CompletableResultCode shutdownResult = null;
 
   LoggerSharedState(
-      Resource resource,
+      Supplier<Resource> resourceSupplier,
       Supplier<LogLimits> logLimitsSupplier,
       LogRecordProcessor logRecordProcessor,
       Clock clock,
       ExceptionAttributeResolver exceptionAttributeResolver) {
-    this.resource = resource;
+    this.resourceSupplier = resourceSupplier;
     this.logLimitsSupplier = logLimitsSupplier;
     this.logRecordProcessor = logRecordProcessor;
     this.clock = clock;
     this.exceptionAttributeResolver = exceptionAttributeResolver;
   }
 
-  Resource getResource() {
-    return resource;
+  // This is used in a test, and must be public for it.
```

Comment:
This is incorrect. All the tests calling this are in the same package so package private will suffice.

Reply:
Ah, yes I think I removed this.

Reply:
Resolved